uefi/src/fs/file_system/error.rs

```diff
@@ -0,0 +1,93 @@
+use crate::fs::{PathBuf, PathError};
+use alloc::string::FromUtf8Error;
+use core::fmt::Debug;
+use derive_more::Display;
+
+/// All errors that can happen when working with the [`FileSystem`].
+///
+/// [`FileSystem`]: super::FileSystem
+#[derive(Debug, Clone, Display, PartialEq, Eq)]
+pub enum FileSystemError {
+    /// Logical errors. See [`LogicError`].
+    Logic(LogicError),
+    /// IO (low-level UEFI-errors) errors. See [`FileSystemIOError`].
+    IO(FileSystemIOError),
+    /// Path-related errors. See [`PathError`].
+    Path(PathError),
+    /// Can't parse file content as UTF-8. See [`FromUtf8Error`].
+    Utf8Encoding(FromUtf8Error),
+}
+
+/// UEFI-error with context when working with the underlying UEFI file protocol.
+#[derive(Debug, Clone, Display, PartialEq, Eq)]
+#[display(fmt = "FileSystemIOError({},{})", context, path)]
+pub struct FileSystemIOError {
```

Comment:
Naming nit: since this is already in the `fs` module, I think we can drop the `FileSystem` prefix from this and `FileSystemIOErrorContext`.

Also, per https://rust-lang.github.io/api-guidelines/naming.html, acronyms a treated as a word for capitalization, so `IoError` instead of `IOError`.

Reply:
Thanks! I've updated the PR